### ISSUE TITLE
Possible Bug Detected: An End Event did not actually end the animation

### ISSUE THREAD

**Issue by wallstop** [KybernetikGames/animancer#152]
# Environment

- Animancer Version Number: v6.0
- Animancer Pro or Lite: Lite
- Unity Version: 2020.2.1f1
- Platform: Windows

# Description

The "Possible Bug Detected: An End Event did not actually end the animation" warning plays with the following code, where we are explicitly clearing the events OnEnd. No other code touches these events.

```csharp
        private void Start()
        {
            AnimancerState state = AnimancerComponent.Play(setupAnimation);
            state.Events.OnEnd += () =>
            {
                OnSetupFinished();
                state.Events.Clear();
            };
        }

        private void OnSetupFinished()
        {
            if (idleAnimation != null)
            {
                _ = AnimancerComponent.Play(idleAnimation);
            }
        }
```

Is there something that we're doing wrong? Is there a better pattern for hooking into OnEnd events?

**Comment by KybernetikGames** [comment]
Setting `state.Events = null;` instead of clearing it would avoid the warning, but if you don't want the event to be triggered every frame then you could just use a regular event instead:

```
state.Events.Add(1, OnSetupFinished);
```

There's also no point in having the `_ = ` before `AnimancerComponent.Play(idleAnimation);`.

**Comment by KybernetikGames** [comment]
This issue is fixed in [Animancer v7.2].
